### PR TITLE
feat:  upgrade llm-d to 0.6

### LINKED ISSUE #5118: Upgrade llm-d to 0.6
/kind feature

**Describe the solution you'd like**

Kserve using llm-d 0.6.

**Anything else you would like to add:**
- https://llm-d.ai/blog/llm-d-v0.5-sustaining-performance-at-scale

### PR DESCRIPTION
**Pull request by pierDipi**
**What this PR does / why we need it**:

 Upgrades llm-d components to 0.6 and adds support for the UDS tokenizer sidecar required by the new precise-prefix-cache-scorer plugin.

  - Bump llm-d-routing-sidecar from v0.4.0 to v0.6.0
  - Bump llm-d-inference-scheduler from v0.4.0 to v0.6.0
  - Bump llm-d-cuda to v0.5.1 (this is the latest and the one that compatible/suggested for v0.6 of other components)
  - Add llm-d-uds-tokenizer (v0.6.0) sidecar container to the scheduler deployment when the precise-prefix-cache-scorer plugin is detected
  - Refactor attachModelArtifacts to accept a target container name and model path, so the tokenizer container can also receive model artifacts (tokenizer files only, via STORAGE_ALLOW_PATTERNS)
  - Update tokenizersPoolConfig in samples to use uds socket instead of tokenizersCacheDir, and require modelName
  - Add new CPU-only simulator sample for precise prefix KV cache routing (llm-inference-service-opt-125m-cpu-kv-cache-routing-simulator.yaml)
  - Fix CreateModelcarContainer to mkdir -p the parent directory before symlinking

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kserve/kserve/issues/5118

**Feature/Issue validation/testing**:

  - Unit tests for isUsingPreciseSchedulingPlugin covering inline JSON/YAML configs, edge cases (nil router, malformed YAML, missing plugins)
  - Unit tests for attachStorageInitializer, attachPVCModelArtifact, and attachOciModelArtifact with tokenizer container targeting
  - E2e test for precise prefix KV cache routing using llm-d-inference-sim simulator (CPU-only, @pytest.mark.llmd_simulator)

- Logs

**Special notes for your reviewer**:

  The tokenizersPoolConfig schema change (dropping tokenizersCacheDir, requiring modelName + uds config) is a breaking change for anyone already using precise-prefix-cache-scorer. Kserve controller handles this change automatically.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the [documentation](https://github.com/kserve/website)?

**Release note**:

```release-note
- Upgrade llm-d to 0.6.
```

### CODE REVIEW

File: pkg/controller/v1alpha2/llmisvc/workload_multi_node.go [hunk 1]
```diff
@@ -202,7 +202,7 @@ func (r *LLMISVCReconciler) expectedMainMultiNodeLWS(ctx context.Context, llmSvc
 			currLeaderSpec = currLWS.Spec.LeaderWorkerTemplate.LeaderTemplate.Spec
 		}
 
-		if err := r.attachModelArtifacts(ctx, serviceAccount, llmSvc, currLeaderSpec, &expected.Spec.LeaderWorkerTemplate.LeaderTemplate.Spec, config); err != nil {
+		if err := r.attachModelArtifacts(ctx, serviceAccount, llmSvc, currLeaderSpec, &expected.Spec.LeaderWorkerTemplate.LeaderTemplate.Spec, config, "main", constants.DefaultModelLocalMountPath); err != nil {
```
Comment (bartoszmajsak) [inline pkg/controller/v1alpha2/llmisvc/workload_multi_node.go:205]: nit, but maybe we can move `main` to const, since it's used in many places.

File: pkg/controller/v1alpha2/llmisvc/workload_storage.go [hunk 2]
```diff
@@ -161,8 +167,8 @@ func (r *LLMISVCReconciler) attachPVCModelArtifact(modelUri string, podSpec *cor
 // Returns:
 //
 //	An error if the configuration fails, otherwise nil.
-func (r *LLMISVCReconciler) attachS3ModelArtifact(ctx context.Context, serviceAccount *corev1.ServiceAccount, llmSvc *v1alpha2.LLMInferenceService, modelUri string, curr corev1.PodSpec, podSpec *corev1.PodSpec, storageConfig *kserveTypes.StorageInitializerConfig, credentialConfig *credentials.CredentialConfig) error {
-	if err := r.attachStorageInitializer(modelUri, curr, podSpec, storageConfig); err != nil {
+func (r *LLMISVCReconciler) attachS3ModelArtifact(ctx context.Context, serviceAccount *corev1.ServiceAccount, llmSvc *v1alpha2.LLMInferenceService, modelUri string, curr corev1.PodSpec, podSpec *corev1.PodSpec, storageConfig *kserveTypes.StorageInitializerConfig, credentialConfig *credentials.CredentialConfig, containerName string, modelPath string) error {
```
Comment (bartoszmajsak) [inline pkg/controller/v1alpha2/llmisvc/workload_storage.go:170]: As a follow-up we might want to think about cleaning signatures a bit, it's getting a bit wild :)

**Review by bartoszmajsak — APPROVED** [review]
/lgtm
/approve

Left two non-blocking comments

File: pkg/controller/v1alpha2/llmisvc/workload_storage.go (hunk 2 above)
Comment (pierDipi) [reply pkg/controller/v1alpha2/llmisvc/workload_storage.go:170]: Agree, I was soo close to doing it but it's a large refactoring

File: hack/setup/infra/manage.istio-helm.sh [hunk 3]
```diff
@@ -91,6 +91,7 @@ install() {
         --namespace "${ISTIO_NAMESPACE}" \
         --version "${ISTIO_VERSION}" \
         --set proxy.autoInject=disabled \
+        --set pilot.env.ENABLE_GATEWAY_API_INFERENCE_EXTENSION=true \
```
Comment (Jooho) [inline hack/setup/infra/manage.istio-helm.sh:94]: Do we use istio for llmisvc from 0.6?

Comment (pierDipi) [reply hack/setup/infra/manage.istio-helm.sh:94]: It can be used for testing but I'm not changing the default, this is just for testing

**Review by Jooho — APPROVED** [review]
/lgtm
/approved
